client/src/app.js

```diff
@@ -44,7 +45,13 @@ class App extends Component {
           <Route exact path="/" component={Home} />
           <Route path="/about" component={About} />
           <Route path="/events" component={Events} />
-          <Route path="/restaurants/:id" component={Restaurant} />
+          <Route path="/restaurants/:id/new-order" component={NewOrder} />
+          <Route
+            path="/restaurants/:id"
+            render={(routeProps) => {
+              return <Restaurant isAuthenticated={isAuthenticated} {...routeProps} />;
+            }}
+          />
```

Comment:
I guess we need change something here. When I work with app 'routes' via app buttons, it is ok. But this is what happens with URL when:
`...restaurants` - renders all Reastaurants
`...restaurants/2` - renders Restaurant with id 2
`...restaurants/2/new-order` - i get this TypeError:
![image](https://user-images.githubusercontent.com/50665431/79635734-57aae700-817b-11ea-9640-7beb91d96e5d.png)

May `BrowserRouter` can help.

Reply:
Fixed. It is SPA, use normal routing